arcgis-ios-sdk-samples/Features/Feature layer selection/FLSelectionViewController.swift

```diff
@@ -19,63 +19,80 @@ class FLSelectionViewController: UIViewController, AGSGeoViewTouchDelegate {
     
     @IBOutlet private weak var mapView:AGSMapView!
     
-    private var map:AGSMap!
-    private var featureTable:AGSServiceFeatureTable!
-    private var featureLayer:AGSFeatureLayer!
-    private var lastQuery:AGSCancelable!
-    private var selectedFeatures:[AGSFeature]!
-    private let FEATURE_SERVICE_URL = "https://sampleserver6.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer/0"
-    
+    private var featureLayer:AGSFeatureLayer?
+    // the query is retained internally by the SDK so use a weak reference
+    private weak var activeSelectionQuery:AGSCancelable?
```

Comment:
It isn't necessary for this to be weak, is there an advantage to making it weak?

Reply:
It's just to illustrate to developers that it doesn't need to be retained by them.